code/game/objects/items/weaponry.dm

```diff
@@ -337,6 +337,32 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	user.visible_message("<span class='suicide'>[user] is slitting [user.p_their()] own throat with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
 	return (BRUTELOSS)
 
+/obj/item/switchblade/makeshift
+	icon_state = "switchblade_ms"
+	desc = "A concealable spring-loaded knife."
+	force = 2
+	throwforce = 3
+	
+/obj/item/switchblade/makeshift/attack_self(mob/user)
+	extended = !extended
+	playsound(src.loc, 'sound/weapons/batonextend.ogg', 50, 1)
+	if(extended)
+		force = 15
```

Comment:
Make three new variables for the blade force, throwforce and icon_state to be set to when extended on `switchblade/attack_self` (values should be 20, 23 and "switchblade_ext"), then set such variables to 15, 15 and "switchblade_ext_ms" on this subtype.
None of this janky pasta please.
See [transforming.dm](https://github.com/Citadel-Station-13/Citadel-Station-13/blob/master/code/game/objects/items/melee/transforming.dm) for reference.

Reply:
Forgive me, but from what I can see the blade force, throwforce is already set 15, and the icon is set correctly to switchblade_ext. Unless im missing something. Im probably missing something.

Reply:
this is the switchblade overhaul i was talking about